### ISSUE TITLE
There is no variable named "terraform" - Terraform Schema

### ISSUE THREAD

**Issue by dtiziani** [hashicorp/vscode-terraform#248]
If I use some kind of interpolation, for instance, to compose the name of a resource depending on the Terraform Workspace, I get this error:

There is no variable named "terraform". Terraform Schema

```
resource "aws_lambda_function" "function_name_lambda" {
  filename      = "./lambda.zip"
  function_name = "function_name${terraform.workspace == "default" ? "" : "_staging"}"
...
```

If I use tflint on console, it doesn't print any errors.

Document reference: 
https://www.terraform.io/docs/state/workspaces.html#current-workspace-interpolation

**Comment by jls-tschanzc** [comment]
I have the same issue in a provider block:
```
provider "aws" {
  profile = "default"
  region  = var.workspace-to-region[terraform.workspace]
  version = "~> 2.40"
}
```
It complains with `There is no variable named "terraform".Terraform Schema` but as dtiziani mentioned, tflint finds no errors and it works fine. I have the language server enabled.

But I am not sure if it just works because of the default fallback value or not.

**Comment by paultyng** [comment]
We just released [v2.0.0-rc.1](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) of the extension. The main features include:

* Added syntax support for 0.12
* Added [terraform-ls](https://github.com/hashicorp/terraform-ls) usage by default (currently on 0.3.0, which offers basic provider code completion)

You can find additional information and specifics in the [release notes](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) and [CHANGELOG](https://github.com/hashicorp/vscode-terraform/blob/master/CHANGELOG.md).

With this release we expect that many of the prior issues and PRs are no longer relevant or have been addressed, and are therefore being closed. If you feel the action taken on an issue or PR is in error, please comment as such and we can figure out the appropriate way to address it.

We plan to add the final 2.0.0 release to the marketplace soon, but are actively seeking your feedback now on the release candidates. You can download the .vsix from the releases page and [manually install it in VS Code](https://code.visualstudio.com/docs/editor/extension-gallery#_install-from-a-vsix) to try it out.
